src/lib/libpthread.js

```diff
@@ -435,11 +435,29 @@ var LibraryPThread = {
         worker = new Worker(p.createScriptURL('ignored'), {{{ pthreadWorkerOptions }}});
       } else
 #endif
-      // We need to generate the URL with import.meta.url as the base URL of the JS file
-      // instead of just using new URL(import.meta.url) because bundler's only recognize
-      // the first case in their bundling step. The latter ends up producing an invalid
-      // URL to import from the server (e.g., for webpack the file:// path).
-      worker = new Worker(new URL('{{{ TARGET_JS_NAME }}}', import.meta.url), {{{ pthreadWorkerOptions }}});
+      {
+#if expectToReceiveOnModule('mainScriptUrlOrBlob')
+        if (Module['mainScriptUrlOrBlob']) {
+          var pthreadMainJs = Module['mainScriptUrlOrBlob'];
+          if (typeof pthreadMainJs != 'string') {
+            pthreadMainJs = URL.createObjectURL(pthreadMainJs);
+          }
+          worker = new Worker(pthreadMainJs, {{{ pthreadWorkerOptions }}});
+        } else
+#endif
+        {
+          if (import.meta.url.startsWith('blob:')) {
+            // If our import.meta.url is a blob, we should just use it.
+            worker = new Worker(import.meta.url, {{{ pthreadWorkerOptions }}});
```

Comment:
I think this syntax is not currently supported by webpack :(   Although it does look like it now is support by vite, which is good!

Reply:
shouldn't there be some mechanism to make emscripten use `worker = new Worker(import.meta.url, ...)`?

i was excited to hear about USE_ES6_IMPORT_META but it seems that's the bundler hack with import.meta.url as the base.